modules/auxiliary/scanner/ssh/ssh_login_pubkey.rb

```diff
@@ -120,6 +122,7 @@ def run_host(ip)
       password: datastore['KEY_PASS'],
       user_file: datastore['USER_FILE'],
       username: datastore['USERNAME'],
+      private_key: datastore['PRIVATE_KEY']
     )
 
     unless keys.valid?
```

Comment:
(Don't seem to be able to comment on an unchanged line; this comment is in reference to Line 137.)

As the change allows running without a `KEY_PATH` set, the log message when running with just a `PRIVATE_KEY` shows `Testing 1 keys from`. Suggest the following to have it read well with the new changes:

```
    key_count = keys.key_data.count                                                                                                                                                                                                  
    key_sources = []                                                                                                                                                                                                                 
    unless datastore['KEY_PATH'].blank?                                                                                                                                                                                              
        key_sources.append(datastore['KEY_PATH'])                                                                                                                                                                                    
    end                                                                                                                                                                                                                              
                                                                                                                                                                                                                                     
    unless datastore['PRIVATE_KEY'].blank?
        key_sources.append('PRIVATE_KEY')  
    end                                                                                                           
                                                         
                                                                                                                  
    print_brute :level => :vstatus, :ip => ip, :msg => "Testing #{key_count} #{'key'.pluralize(key_count)} from #{key_sources.join(' and ')}"
```